### DISCUSSION TITLE
Is it possible to make api /openid-connect/auth to NOT expire cookie KC_RESTART when user hasn't logged in yet?

### DISCUSSION THREAD

**Discussion by quyhx** [keycloak/keycloak#9518] [Q&A]
We are using Keycloak version 13.0.1 as an SSO system for our platforms. I assume that our main domain is abc.com, and all platforms are using subdomain xxxxx.abc.com. Currently, we are encountering an issue as below:
1. At the browser, tab 1, user opens keycloak SSO login page which is under **account.abc.com (1)**.  This is login page of Keycloak. By default, this login page is initialized with cookie **KC_RESTART**.
2. At the same browser, tab 2, user opens our website under **web.abc.com (2)** which is using (1) as an authentication server. This website  is using Angular OIDC library. When this website is initialized, OIDC lib calls `/openid-connect/auth`  of (1) to check user state. As I can see now, since user is not in logged-in state, then the api clears authenticate session and make cookie **KC_RESTART** of (1) to be expired.
3. As a result, when user back to tab 1 which is opened at step 1 and try to login, user encounters an error since system cannot find authentication session and then try to use **KC_RESTART** cookie to reset authentication session however the cookie is even not there since it is expired at step 2.

So my question is: Is there any way to make api `/openid-connect/auth` NOT expire KC_RESTART cookie when user is not in logged-in state? or any way to re-init authentication session or KC_RESTART cookie when trying to login or any other way to make login can go through without any error for this case?

The detail of APIs:
1. /auth API is: `https://{{account.abc.com}}/auth/realms/{{real_name}}/protocol/openid-connect/auth?response_type=code&client_id={{client_id}}&state={{state}}&redirect_uri={{redirect_uri}}&scope=openid&code_challenge={{code_challenage}}&code_challenge_method=S256&nonce={{nonce}}&prompt=none`
2. login action: `https://{{account.abc.com}}/auth/realms/{{real_name}}/login-actions/authenticate?session_code={{session_code}}&execution={{execution_code}}&client_id={{client_id}}&tab_id={{tab_id}}`

Please let me know if anything is not clear, so that I can explain further.

**Comment by yakyouk** [comment]
I have the same issue. Specifically, when a user doesn't have credentials, my app occasionally checks auth status in case the user authenticated externally. If this check happens while a user has the login page open, the user won't be able to complete auth.

To reproduce:
- in tab 1, open auth page: `/auth/realms/my-realm/protocol/openid-connect/auth?response_type=code&client_id=my-client&state=abc1234&scope=profile&code_challenge=def5678&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fexample.com%2Fsignin%3Faction%3Dclose
- in tab 2, to simulate an authentication check, open the following: `/auth/realms/my-realm/protocol/openid-connect/auth?prompt=none&response_type=none&client_id=my-client&code_challenge=def5678&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fexample.com%2Fsignin%3Faction%3dmessage`. If this redirects to `redirect_url` with `error=login_required`, this tells us the user isn't authenticated. Note the `prompt=none` and `response_type=none`.
- return to tab 1 and complete authentication, and see error `Cookie not found. Please make sure cookies are enabled in your browser.`

I propose that the cookie not be cleared if `response_type=none`.

Opened issue #11340

**Comment by edmilsonss** [comment]
Hi everyone. I have a similar situation.

I have a Keycloak 18.0.2 integrated to an external IdP (Okta / Ping). I see lots of IDENTITY_PROVIDER_LOGIN_ERROR / cookie_not_found in my login events.

The steps to reproduce also are the same and also include stale sessions if they are between Session Idle and Session Max.

In the client application the user is only able to access if he clears the browser cookies and cache.

Any thoughts about this?

**Reply by edmilsonss** [reply]
My clue is that this error seems to be related to multiple use cases.

In my case, changing the client application to handle "login_required" and "invalid session" and try to do a silent signin or do a signin redirect if signin silent fails. This seems do be effective.

Does this make sense or is justa a workaround?

In my use case the client application uses a oidc-client-ts lib.
